### DISCUSSION TITLE
Custom math string post processor

### DISCUSSION THREAD

**Discussion by Kazbek** [KaTeX/KaTeX#3933] [Q&A]
Im using math like this e=mc^2 and not it output to 
```
<span class="math">\(e=mc^2\)</span>
``` 
and want to customize output Latex formula. Not just place it "as is" but  process with my string to string function

```
string processTexFormula(string rawTex)
{
    return"test_" + rawText;
}
```

so output will be:
```
<span class="math">\(test_e=mc^2\)</span>
```

Can I somehow configure Math processor?
I guess it must affect in this two places: [one](https://github.com/xoofx/markdig/blob/fcb56fb03701cc1a07a52769908579adb8927ee2/src/Markdig/Extensions/Mathematics/HtmlMathInlineRenderer.cs#L25) and [two](https://github.com/xoofx/markdig/blob/fcb56fb03701cc1a07a52769908579adb8927ee2/src/Markdig/Extensions/Mathematics/HtmlMathBlockRenderer.cs#L25)

**Comment by edemaine** [comment]
I'm confused — how are you calling KaTeX? The two links you gave are not KaTeX code, but other things that presumably call KaTeX, but I don't see the call.

If you're using one of [our extensions](https://katex.org/docs/libs), I'm happy to help (e.g. auto-render has a `preProcess` callback), but otherwise, I don't know how they work so can't really help. Maybe ask in another forum?

**Reply by Kazbek** [reply]
I want to prerender it on server by passing this string into `katex.renderToString` on js interpretator [Jint](https://github.com/sebastienros/jint). So `processTexFormula` will return katex html output.

**Reply by edemaine** [reply]
If you'd like to automatically convert `\(...\)` passages into math, then use [auto-render's renderMathInText](https://github.com/KaTeX/KaTeX/blob/3d5de92fb0d0511ac64901bb60b5d46c5677eb28/contrib/auto-render/auto-render.js#L9), which supports a `preProcess` option. If you'd like to look for `<span class="math">`, I'd recommend using a regular expression `replace` and calling `katex.renderToString` after modifying the string how you'd like.

**Reply by Kazbek** [reply]
`renderMathInText` looks like nice fast solution, I will try it.
